frontend/src/pages/EditRolesPage/components/MemberTable/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect } from 'react';
+import { format, parseISO } from 'date-fns';
+
+import EditRolesButton from '../buttons/EditRolesButton';
+
+import { Loading, RequestErrorSnackbar, Table } from '@SharedComponents';
+import { useRequest } from '@Hooks';
+import { getMultipleInductionClasses } from '@Services/InductionClassService';
+import {
+  MultipleInductionClassResponse,
+  InductionClassResponse,
+  AppUserResponse,
+  MultipleAppUserResponse,
+} from '@Services/api';
+import { inductionClassDateFormat } from '@Constants/dateTimeFormat';
+import { getMultipleUsers } from '@Services/UserService';
+
+// Do Use Effect so table reloads
+
+const appUserResponseToAppUserRow = ({
+  id,
+  firstName,
+  lastName,
+  role,
+  email,
+}: AppUserResponse) => {
+  const userRow = {
+    id,
+    firstName,
+    lastName,
+    name: firstName + ' ' + lastName,
+    email,
+    role,
+  };
+
+  return userRow;
+};
+
+function MemberTable(): JSX.Element {
+  const { data, isLoading, error, isError } = useRequest<
+    MultipleAppUserResponse,
+    Response
+  >({
+    requestKey: 'getMultipleUsers',
+    requestFunc: getMultipleUsers,
+    requestParams: [{ showAffiliates: false }],
+  });
+
+  const columns = [
+    { title: 'Name', field: 'name' },
+    { title: 'Email', field: 'email' },
+    { title: 'Role', field: 'role' },
+    {
+      title: '',
+      render: ({ id, firstName, lastName, email, role }: AppUserResponse) => (
+        <EditRolesButton
+          id={id}
+          email={email}
+          firstName={firstName}
+          lastName={lastName}
+          role={role}
+        />
+      ),
+    },
+  ];
+
+  const ComponentToRender = () => {
```

Comment:
it's been a while, but can't we just return the function instead of naming it `ComponentToRender?`

Reply:
I'll check this out more in detail and will follow up. I believe initially I had some issues with referencing the types from the MultipleAppUserResponse imports when just returning the function but worked when putting it under a component. Again, I'll look more deep into this.